pkg/agent/openflow/pipeline.go

```diff
@@ -1337,14 +1337,16 @@ func (c *client) l3FwdServiceDefaultFlowsViaGW(ipProto binding.Protocol, categor
 	gatewayMAC := c.nodeConfig.GatewayConfig.MAC
 
 	flows := []binding.Flow{
-		// This flow is used to match the packets of Service traffic:
-		//	- NodePort/LoadBalancer request packets which pass through Antrea gateway and the Service Endpoint is not on
+		// This generates the default flow to match Service packets, traffic cases include:
+		//  - When Egress is not enabled:
+		//    - Service request packets sourced from Antrea gateway and destined for external Endpoint.
+		//    - Service request packets sourced from local Pods and destined for external Endpoint.
 		//    local Pod CIDR or any remote Pod CIDRs.
-		//	- ClusterIP request packets which are from Antrea gateway and the Service Endpoint is not on local Pod CIDR
-		// 	or any remote Pod CIDRs.
-		//  - NodePort/LoadBalancer/ClusterIP response packets.
-		// The matched packets should leave through Antrea gateway, however, they also enter through Antrea gateway. This
-		// is hairpin traffic.
+		//  - When Egress is enabled:
+		//    - Service request packets sourced from Antrea gateway and destined for external Endpoint.
+		// Note that, when Egress is enabled, Service request packets sourced from local Pods and destined for external
+		// Endpoint should not be matched by the flow, and such packets should be matched by the flow with higher priority
+		// in L3ForwardingTable, which is installed by Egress.
```

Comment:
This two flow have the same priority now as the following. But in my testbed, if the first one is installed, the all packets from local Pods are matched by the flow. However, I think there is risk that the second flow matches the packets from local Pods since they have the same priority 190.
```
 cookie=0x45060000000000, table=70, priority=190,ct_state=-rpl+trk,ip,reg0=0x2/0xf actions=resubmit(,71)
 cookie=0x45040000000000, table=70, priority=190,ct_state=+trk,ct_mark=0x10/0x10,ip,reg0=0x80000/0x80000,reg4=0/0x200000 actions=mod_dl_dst:aa:c3:42:b5:e8:7e,resubmit(,72)
```
I'll change the comment back and leave a TODO. We can have a separate PR for the issue.